File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i:nums){
            map.put(i,map.getOrDefault(i,0)+1);
        }
        int n=nums.length/2;
        // System.out.println(map);
        for(Map.Entry<Integer,Integer> hm:map.entrySet()){
            if(n<hm.getValue()){
                return hm.getKey();
            }
        }
        return -1;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n=int(len(nums)/2)
        freq={}
        for i in nums:
            if i in freq:
                freq[i]+=1
            else:
                freq[i]=1
        print(freq,n)
        for key,value in freq.items():
            if n<value:
                return key